From 60e5a6bbf4172079fb72aadbc3021a3de3cf937e Mon Sep 17 00:00:00 2001
From: Marek Blaha <mblaha@redhat.com>
Date: Fri, 3 Jan 2025 15:45:12 +0100
Subject: [PATCH 08/24] dnfdaemon: Properly leave event loop

Exit the event loop and properly join the serving thread.
---
 dnf5daemon-client/main.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dnf5daemon-client/main.cpp b/dnf5daemon-client/main.cpp
index bab0243b..e759bd33 100644
--- a/dnf5daemon-client/main.cpp
+++ b/dnf5daemon-client/main.cpp
@@ -292,6 +292,8 @@ int main(int argc, char * argv[]) {
 
         // Run selected command
         command->run();
+
+        connection->leaveEventLoop();
     } catch (libdnf5::cli::ArgumentParserError & ex) {
         std::cerr << ex.what() << _(". Add \"--help\" for more information about the arguments.") << std::endl;
         return static_cast<int>(libdnf5::cli::ExitCode::ARGPARSER_ERROR);
-- 
2.48.1

